lib/ecto/changeset.ex

```diff
@@ -2929,7 +2929,11 @@ end
 defimpl Inspect, for: Ecto.Changeset do
   import Inspect.Algebra
 
-  def inspect(changeset, opts) do
+  def inspect(%Ecto.Changeset{changes: changes, data: data} = changeset, opts) do
+    changeset = case data do
+      %type{} -> %Ecto.Changeset{changeset | changes: filter(changes, type.__schema__(:redacted_fields))}
+      _ -> changeset
+    end
```

Comment:
Instead of applying the filter here, maybe we can apply it a couple lines below where we match on {:changes, changes}?

Reply:
We want to redact `data` as well, right?

Reply:
`data` will be redacted as part of the inspect protocol though. I will look into a separate PR that automatically derives inspect based on redacted if one was not derived yet.

Reply:
Yes, I went through all the work to write tests and redact values in data also, and then the tests verifying that it didn't redact when not set failed because the password still didn't show 😂